### PR TITLE
Allow passing the icon name as a positional argument to `OcticonComponent`

### PR DESCRIPTION
**Pull request by manuelpuyol**
Closes https://github.com/primer/view_components/issues/394

This will allow:
```rb
Primer::OcticonComponent.new(icon: :x)
Primer::OcticonComponent.new(:x)
primer_octicon(icon: :x)
primer_octicon(:x)
```

### CODE REVIEW

**Comment by srt32** [comment]
The driver here is a less verbosity when using the helper?

**Comment by manuelpuyol** [comment]
> The driver here is less verbosity when using the helper?

yes!

File: app/components/primer/octicon_component.rb [hunk 1]
```diff
@@ -18,19 +18,20 @@ class OcticonComponent < Primer::Component
...
-    #   <%= render(Primer::OcticonComponent.new(icon: "x", size: :large)) %>
+    #   <%= render(Primer::OcticonComponent.new("x", size: :large)) %>
     #
     # @param icon [String] Name of [Octicon](https://primer.style/octicons/) to use.
     # @param size [Symbol] <%= one_of(Primer::OcticonComponent::SIZE_MAPPINGS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-    def initialize(icon:, size: SIZE_DEFAULT, **system_arguments)
-      @icon = icon
+    def initialize(icon_name = nil, icon: nil, size: SIZE_DEFAULT, **system_arguments)
+      @icon = icon_name || icon
```
Comment (srt32) [inline app/components/primer/octicon_component.rb:34]: Do we plan to keep this support forever or think we'll deprecate the old way at some point?

**Review by srt32 — APPROVED** [review]

File: app/components/primer/octicon_component.rb (hunk 1 above)
Comment (manuelpuyol) [reply app/components/primer/octicon_component.rb:34]: we can deprecate it at some point

**Pull request merged**